libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.policy;
+
+import org.elasticsearch.entitlement.runtime.api.ElasticsearchEntitlementChecker;
+import org.elasticsearch.entitlement.runtime.api.NotEntitledException;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class PolicyManager {
+    private static final Logger logger = LogManager.getLogger(ElasticsearchEntitlementChecker.class);
+
+    protected final Map<String, List<Entitlement>> mainPolicies;
+    protected final Map<String, Policy> pluginPolicies;
+    private final Function<Class<?>, String> pluginResolver;
+
+    public static final String ALL_UNNAMED = "ALL-UNNAMED";
+
+    public PolicyManager(Policy defaultPolicy, Map<String, Policy> pluginPolicies, Function<Class<?>, String> pluginResolver) {
+        this.mainPolicies = Objects.requireNonNull(defaultPolicy).scopes.stream()
```

Comment:
Why decompose the server policy, couldn't we keep it as a Policy object?

Reply:
It will be clear later, but you are right, at this point is not needed. I'll revert this, and decompose it when needed.